### PR TITLE
[WIP] Azure IoT test

### PR DESCRIPTION
**Pull request by dautovri**
Azure IoT Edge test  based on qcow2 with multiple disks 
Signed-off-by: Ruslan Dautov <dautovri@hpcdlab.com>

### CODE REVIEW

File: tests/azure_iot/testdata/azure_iot.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+[!exec:grep] stop
+[!exec:cat] stop
+[!exec:curl] stop
+
+exec docker run  --name azure_iot  dautovri/eden-azure-iot-test
+exec docker cp  azure_iot:/disk  disk/
+exec docker rm -f azure_iot
+
+eden -t 10m pod deploy -p 8028:80 --vnc-display=1 file://disk/Ubuntu-18.04-bionic --disks=file://disk/Ubuntu-18.04-bionic-1 --disks=file://disk/Ubuntu-18.04-bionic-2 --disks=file://disk/Ubuntu-18.04-bionic-3 -n azureiot --memory 3GB -v debug
+stdout 'deploy pod azureiot with ubuntu qcow2 disks request sent'
```
Comment (giggsoff) [inline tests/azure_iot/testdata/azure_iot.txt:11]: Here we should use something like `stdout 'deploy pod azureiot.*request sent'`. Current variant not works for me.

File: tests/azure_iot/testdata/azure_iot.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    eve:
+      {{EdenConfig "eve.name"}}:
+        onboard-cert: {{EdenConfigPath "eve.cert"}}
+        serial: "{{EdenConfig "eve.serial"}}"
+        model: {{EdenConfig "eve.devmodel"}}
+
+
+-- get.sh --
+address=`grep "^$1"'\s*' pod_ps | cut -f 5`
+until curl $address; do sleep 5; done
```
Comment (giggsoff) [inline tests/azure_iot/testdata/azure_iot.txt:37]: In my experiments, application do not respond on 80 port. As I understand it is expected behavior. 
Can we move our test onto testing logs of our application? I mean, we can try to find something inside `./eden pod logs --fields=app --format=json azure_iot`?

**Review by giggsoff — CHANGES_REQUESTED** [review]
Please take a look onto my comments.

File: tests/azure_iot/testdata/azure_iot.txt (hunk 1 above)
Comment (dautovri) [reply tests/azure_iot/testdata/azure_iot.txt]: Changed, on  `test eden.app.test -test.v -timewait 20m RUNNING azureiot`

File: tests/azure_iot/testdata/azure_iot.txt (hunk 2 above)
Comment (dautovri) [reply tests/azure_iot/testdata/azure_iot.txt]: Changed on` ./eden pod logs --fields=app --format=json azure_iot ` and grep

File: tests/azure_iot/testdata/azure_iot.txt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+[!exec:bash] stop
+[!exec:grep] stop
+[!exec:cat] stop
+[!exec:curl] stop
+
+exec docker run  --name azure_iot  dautovri/eden-azure-iot-test
```
Comment (mydatascience) [inline tests/azure_iot/testdata/azure_iot.txt:6]: Why do we need it? To create a local image? 

I'd rather not do "exec docker run" in tests

File: tests/azure_iot/testdata/azure_iot.txt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+exec docker cp  azure_iot:/disk  disk/
+exec docker rm -f azure_iot
+
+eden -t 10m pod deploy file://disk/Ubuntu-18.04-bionic --disks=file://disk/Ubuntu-18.04-bionic-1 --disks=file://disk/Ubuntu-18.04-bionic-2 --disks=file://disk/Ubuntu-18.04-bionic-3 -n azureiot --memory 3GB -v debug
```
Comment (mydatascience) [inline tests/azure_iot/testdata/azure_iot.txt:10]: 3 ubuntu ??? it's not clear why

**Review by mydatascience — CHANGES_REQUESTED** [review]

File: tests/azure_iot/testdata/azure_iot.txt (hunk 3 above)
Comment (mydatascience) [reply tests/azure_iot/testdata/azure_iot.txt:6]: Also we'd like to have the Dockerfile  attached

Comment (rvs) [reply tests/azure_iot/testdata/azure_iot.txt:6]: This remind me -- we should NEVER be using private docker images. All images should come from either lfedge or itmo orgs @mydatascience @dautovri

Comment (giggsoff) [reply tests/azure_iot/testdata/azure_iot.txt:6]: @dautovri I pushed you image into itmoeve/eden-azure-iot-test.

**Pull request closed without merging**